src/common/src/types/to_sql.rs

```diff
@@ -74,3 +76,21 @@ impl ToSql for JsonbRef<'_> {
         Ok(IsNull::No)
     }
 }
+
+impl ToSql for ListRef<'_> {
+    to_sql_checked!();
+
+    fn accepts(ty: &Type) -> bool
+    where
+        Self: Sized,
+    {
+        matches!(ty.kind(), postgres_types::Kind::Array(_))
+    }
+
+    fn to_sql(&self, ty: &Type, out: &mut BytesMut) -> Result<IsNull, Box<dyn Error + Sync + Send>>
+    where
+        Self: Sized,
+    {
+        self.iter().collect_vec().to_sql(ty, out)
```

Comment:
Make sure only 1 dimension?

Reply:
https://docs.rs/postgres-types/0.2.6/postgres_types/trait.ToSql.html#arrays

https://github.com/risingwavelabs/risingwave/blob/855cd7d9037fa0574d8c9dec4c8e2c1621c9f4f8/src/common/src/types/to_binary.rs#L124-L127
I think it should be guaranteed here that there will only be an array of 1 dimension?

Reply:
But as a general trait implementation, `to_sql` may have other callers (in the future) in addition to `to_binary`.

Reply:
The link above is for `FromSql`. There seems to be no check of multiple dimensions in `ToSql` there:
https://github.com/sfackler/rust-postgres/blob/e528e01bbe2f7e198a55112fd6de3a24db3b8a6e/postgres-types/src/lib.rs#L960

Can we confirm by actually executing `to_sql` on a `Vec<Vec<i32>>` or `ListRef` of `ListRef` of `i32` value?